core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -464,6 +464,99 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
   ): SqlStreamingAction[Seq[Entity], Entity, Read] =
     queryEntitiesWithFilter(workspaceId, entityType, entityQuery, sql"")
 
+  /**
```

Comment:
written by AI except for sql that figures out what needs to be updated in the attributes json and updates it